nixos/modules/services/continuous-integration/woodpecker/agent.nix

```diff
@@ -0,0 +1,78 @@
+{
+  config
+  , lib
+  , pkgs
+  , ...
+}:
+
+let
+  cfg = config.services.woodpecker-agent;
+in {
+  meta.maintainers = [ lib.maintainers.janik ];
+
+  options = {
+    services.woodpecker-agent = {
+      enable = lib.mkEnableOption (lib.mdDoc "Woodpecker-Agent executes tasks from the Woodpecker-Server");
+      package = lib.mkPackageOptionMD pkgs "woodpecker-agent" { };
+      stateDir = lib.mkOption {
+        default = "/var/lib/woodpecker";
+        type = lib.types.str;
+        description = lib.mdDoc "woodpecker data directory.";
+      };
```

Comment:
Will have to look at the StateDir option, since the module was quiet heavily inspired by the gitea service and they weren't using it I didn't look at that to much. Don't know what you mean with the tmpfiles entries. DynamicUser shouldn't matter since all the code the pipeline executes get executed in oci containers with for example docker or podman.

Reply:
Okay so when having statedir enabled and setting workdir to /var/lib/${statedir} the service can write, is that actually wanted? I looked at some other service definitions and found that they are doing it in the same way. Doesn't that take away the users flexibility to store stuff anywhere else then /var/lib/service?